apps/schools/domains/common/components/containers/BaseLayoutComponents/Menu/index.tsx

```diff
@@ -31,33 +31,30 @@ const MenuCustom = () => {
         setConditions({ isOrganizationSelected: organization.id !== undefined, permanentDisabled: false })
     }, [organization])
 
+    const memoizedMenuItems = useMemo(() => {
+        return menuList.map((el) => (
+            <Menu.Item disabled={el.isDisabled(conditions)} key={el.url} icon={el.icon} className={styles.menuItem}>
+                {el.name}
+            </Menu.Item>
+        ))
+    }, [conditions])
+
     return (
-        <>
-            <Menu
-                className={styles.menu}
-                theme='light'
-                mode='inline'
-                onClick={(e) => {
-                    if (!router.asPath.endsWith(e.key)) {
-                        router.push(`/${e.key}`)
-                    }
-                }}
-                selectedKeys={menuList
-                    .filter((el) => router.asPath.includes(el.url) && !el.isDisabled(conditions))
-                    .map((el) => el.url)}
-            >
-                {menuList.map((el) => (
-                    <Menu.Item
-                        disabled={el.isDisabled(conditions)}
-                        key={el.url}
-                        icon={el.icon}
-                        className={styles.menuItem}
-                    >
-                        {el.name}
-                    </Menu.Item>
-                ))}
-            </Menu>
-        </>
+        <Menu
+            className={styles.menu}
+            theme='light'
+            mode='inline'
+            onClick={(e) => {
+                if (!router.asPath.endsWith(e.key)) {
+                    router.push(`/${e.key}`)
+                }
+            }}
+            selectedKeys={menuList
+                .filter((el) => router.asPath.includes(el.url) && !el.isDisabled(conditions))
+                .map((el) => el.url)}
```

Comment:
Perhaps I would combine these two methods into one to avoid double traversal of the array. Right now I've read how to do it - use "reduce". I do not know how correct this is.